engines/osxnative/sandboxbuilder.go

```diff
@@ -0,0 +1,33 @@
+package osxnative
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type sandboxbuilder struct {
+	engines.SandboxBuilderBase
+	env         map[string]string
+	taskPayload *payload
+	context     *runtime.TaskContext
+}
+
+func (s sandboxbuilder) SetEnvironmentVariable(name string, value string) error {
+	_, ok := s.env[name]
+	if ok {
+		return engines.ErrNamingConflict
+	}
+
+	s.context.Log("Setting \"", name, "\" environment variable to \"", value, "\" ")
+	s.env[name] = value
+	return nil
+}
+
+func (s sandboxbuilder) StartSandbox() (engines.Sandbox, error) {
+	env := []string{}
+	for name, value := range s.env {
+		env = append(env, name+"="+value)
+	}
+
+	return newSandbox(s.context, s.taskPayload, &env), nil
```

Comment:
Do you see a time where we might have other configuration options fo rthe sandbox that we want to pass in? Perhaps there could be a sandbox configuration object we pass in, and env is part of that.  Then newSandbox knows what to do with that config that we pass in.